src/relay/transforms/annotate_target.cc

```diff
@@ -77,7 +77,11 @@ class AnnotateTargetRewriter : public ExprRewriter {
         compiler_ends.push_back(call->args[0]);
       } else if (op_expr_to_target_.find(arg) != op_expr_to_target_.end()) {
         arg_target = op_expr_to_target_[arg];
-        compiler_ends.push_back(InsertAnnotation(arg, arg_target, make_end_op));
+        if (call && call->args.size() == 0) {
```

Comment:
Please add comments to indicate in which case we don't want `compiler_end`.

Reply:
Done.

Reply:
Thanks. Would you be more specific like "If an argument is a call node and has no argument, then it should be tensor ops such as zeros, so we treat it as input vars."

Reply:
Done.